client/authority-discovery/src/worker/addr_cache.rs

```diff
@@ -17,98 +17,109 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use libp2p::core::multiaddr::{Multiaddr, Protocol};
-use std::collections::HashMap;
 
 use sc_network::PeerId;
 use sp_authority_discovery::AuthorityId;
+use std::collections::{hash_map::Entry, HashMap, HashSet};
 
 /// Cache for [`AuthorityId`] -> [`Vec<Multiaddr>`] and [`PeerId`] -> [`AuthorityId`] mappings.
```

Comment:
Just nitpicking here: Either remove concrete types from the struct description, or change them to the current ones.